.github/workflows/build-test.yml

```diff
@@ -0,0 +1,55 @@
+# This workflow uses actions that are not certified by GitHub.
+# They are provided by a third-party and are governed by
+# separate terms of service, privacy policy, and support documentation.
+# This workflow will do a clean install of dotnet dependencies
+
+name: Build and Test
+
+on:
+  push:
+    branches: [ '**' ]
+  pull_request:
+    branches: [ master ]
+
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+jobs:
+  build_test:
+    name: Build and Test on dotnet ${{ matrix.dotnet-version }} and ${{ matrix.os }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        dotnet-version: ['2.1.x']
+        os: [ubuntu-latest]
```

Comment:
Should we be doing latest here? should we lock to a particular version?

Reply:
I tried it didn't work.  I think we would have to change some of the .csproj files.  I will create another issue to look into this.